pkg/cluster/kubernetes.go

```diff
@@ -68,7 +68,7 @@ func (k *KubernetesClient) K8sRestConfig(ctx context.Context) (*rest.Config, err
 	config.Timeout = time.Minute
 
 	if k.ForceEndpoint != "" {
-		config.Host = fmt.Sprintf("%s:%d", k.ForceEndpoint, constants.DefaultControlPlanePort)
+		config.Host = fmt.Sprintf("%s:%d", k.ForceEndpoint, 7444 /*constants.DefaultControlPlanePort*/)
```

Comment:
Unsure how to properly plumb `controlPlanePort` from cluster create args into here

Reply:
I'd think we need to add a flag here which tells the control plane port, or even a full hostname:port pair

Reply:
`--control-plane-port` does override that, but port is not plumbed in here thus waiting for k8s pods fails due to failure to dial port 6443

Reply:
hold on, these are different

* `--control-plane-port` is the port API server listens on (in this case, _inside_ the container) (defaults to 6443)
* there's also internal LB (KubePrism) which is not accessible outside of the cluster (defaults to 7445)
* there's a port we mapped the `--control-plane-port` to outside of the cluster -- this should go into the `kubeconfig`

What should happen is:  `talosctl cluster create` in Docker mode should create a Docker port mapping from a random available host port (e.g. 34321) to `--control-plane-port` (e.g. 6443) in the container. Inside the container nothing changes, but outside the container `kubeconfig` points to `127.0.0.1:34321` now. So the `pkg/cluster` code needs a hint to know which one if the external endpoint, it could be a full hostname:port pair as well.

Reply:
`--control-plane-port` also affects to where the port is mapped, check https://github.com/siderolabs/talos/pull/8412/files#diff-6ed6be17858f4d4a230f75c46a3ade864e225e78d87760e13788db4bb58fe534R778

Reply:
this is something we need to change and map to a random available localhost port

the whole goal of the issue is to find two available ports on localhost, and then map Talos API and Kubernetes API to those ports.

Reply:
yes, but it doesn't work in the current condition as of main due to this preference not being forwarded to the status checkers

Reply:
see above - we need to implement a proper way to pass the new endpoint down the stack, one of the options I suggested is to accept full `hostname:port` pair as on override